Service/DataService.php

```diff
@@ -78,6 +78,8 @@ class DataService
     protected $instanceId;
     /** @var array */
     private $cacheBusinessKey = [];
+    /** @var array */
+    private $cacheOuuids = [];
```

Comment:
Service should be stateless. Instead of saving the Ouuids in a property. Return them in a Ouuids Cache object? Then pass it everywhere needed.

Reply:
I left Todos (it's the same for cacheBusinessKey) but for this we should use a cache service (redis?) 19413cae6d198d00391e9ef98dcd65d25c2950d8

Reply:
Don't think we need redis for this. I think of redis in terms as message queue, the caching needed here can be handled in a php object (Im ok with the todo's --> IDEALY no todo but a ticket (you can create one that references the line number)

Reply:
https://github.com/ems-project/EMSCoreBundle/issues/316